boa/src/syntax/lexer/cursor.rs

```diff
@@ -113,8 +175,11 @@ where
         };
 
         match chr {
-            Some(Ok('\r')) => self.carriage_return(),
-            Some(Ok('\r')) | Some(Ok('\u{2028}')) | Some(Ok('\u{2029}')) => self.next_line(),
+            Some(Ok('\r')) => {
+                self.carriage_return();
+                self.next_line()
```

Comment:
Why was this changed? A carriage return does not change the line number, AFAIK. Also, a `next_line()` call also does a carriage return, so the first carriage return wouldn't make sense.

Reply:
> Why was this changed? A carriage return does not change the line number, AFAIK. Also, a `next_line()` call also does a carriage return, so the first carriage return wouldn't make sense.

I took this table in the spec to indicate a carriage return was a line terminator https://tc39.es/ecma262/#sec-line-terminators perhaps I misunderstood. 

I didn't notice the next_line did a carriage return so yeah that can be fixed

Reply:
The thing is that even if a `\r` is considered a "line terminator token", when we report lines, we have to take into account how do editors show this. I think a `\r` character just moves the current column to number `1`.